### PR TITLE
0.13 improvement(security): reference all images by sha256 from garden

### PR DESCRIPTION
**Pull request by stefreak**
**What this PR does / why we need it**:
- improvement: delete unused images
- improvement: remove parametrization from buildkit image
- ci: fix test-docker-gcloud
- improvement: emphasize we do not publish alpine-builder
- docs: use recent gcloud image tag
- improvement(security): reference all builtin images using sha256 digest
- improvement(security): add sha256 hashes to gardendev images docker files

**Which issue(s) this PR fixes**:

Addresses #3993 

**Special notes for your reviewer**:

### CODE REVIEW

File: images/buildkit/Dockerfile [hunk 1]
```diff
@@ -14,7 +12,12 @@ RUN curl -fsSL "https://github.com/GoogleCloudPlatform/docker-credential-gcr/rel
   | tar xz --to-stdout ./docker-credential-gcr \
   > /tmp/docker-credential-gcr && chmod +x /tmp/docker-credential-gcr
 
-FROM moby/buildkit:${BUILDKIT_VERSION}${BASE_IMAGE_SUFFIX} as output
+FROM moby/buildkit:v0.10.5 as buildkit
```
Comment (TimBeyer) [inline images/buildkit/Dockerfile:15]: What about these images?

Comment (stefreak) [reply images/buildkit/Dockerfile:15]: You are right, will add hashes here as well.

Comment (stefreak) [reply images/buildkit/Dockerfile]: Done 👍

**Review by TimBeyer — APPROVED** [review]

**Comment by stefreak** [comment]
Test failure in vm-microk8s is also happening in 0.13, so we did not introduce it. Merging this despite failures.

**Pull request merged**